### ISSUE TITLE
Tasks cannot be completed / modified on mobile browsers - incorrect mouseover/click behaviour

### ISSUE THREAD

**Issue by Alys** [HabitRPG/habitica#11181]
@kugoi (ba58893d-a44a-47d6-9d1a-16b4ac8c58c8) reports:

**"On Android, viewing task list on Chrome in desktop mode, the broken challenge link no longer brings up the box to keep or remove tasks and the 3 dot more options won't bring up the options**
**"mobile chrome on desktop mode.**
**"Same problem when not in desktop mode"**

I can't replicate the 3-dots menu problem in Firefox on my Android phone (although it's fiddly to click the right place for the three-dots menu).

I can't replicate this on Chrome on my phone because of this bug: https://github.com/HabitRPG/habitica/issues/11180 ("can't log in to Habitica website on non-default mobile browser").

Chris Moschini   @b9chris (3453f276-65dd-4736-ace5-28d5ab484957) reports:
**"Cannot Check Off Tasks with Touchscreen**
**"I'm on a Win10 touchscreen (Surface Book), and tapping a checkbox to check it off with Touch fails to check off the box. It acts like a mouseover. Out of curiosity, double tapping, or tapping many times slowly, still doesn't check off the box. It just triggers what mousing over (but not clicking) would do were I using a mouse."**

**Comment by kugoi** [comment]
I realized after Chris posted that I can't check tasks as done either.

**Comment by b9chris** [comment]
I'm a Web Dev and happy to work on this, but, it appears there are a lot of hurdles to becoming a HabitRPG dev, so I'm not clear on how long that would take to get through, and, it would help if an experienced dev could pass me a guess at where this would be. If it's jQuery based then honestly the fix is to just change .on('click' ... to .on('click touch', but, it depends on how this is written.

**Comment by citrusella** [comment]
For what it's worth, I tried this in Firefox (reasonably up-to-date, I assume) on a Samsung Android tablet (not phone) a couple days ago after seeing the reports in-guild and didn't have any of the issues, though for what it's worth I also didn't have any small screen behavior due to the tablet's size. (I did have trouble hitting the challenge link for a bit when I first tried but that was more due to its size.)

The issues also don't seem to occur on my (HTC Desire 526, Marshmallow I think) phone's less-up-to-date Firefox (53 as opposed to the current 66 or so). Technically I can't hit the three dots reliably, I guess, but that's because when I tried tapping in their general vicinity long enough initially they disappeared from all tasks for some reason... they would briefly reappear if I pressed the broken challenge link (which was for some reason easier for me to click here despite its even smaller size...) and I could effectively get them to toggle open and closed at that point but couldn't get them to reliably reappear on all tasks again? For all I know I have to clear the cache but I'm not quite sure why they disappeared in the first place????

TL;DR: I can get all three things to properly happen on different size formats of the website on different devices, though the smaller the screen, the weirder the behavior, it seems.

**Comment by kugoi** [comment]
I'm using Chrome, so...

**Comment by b9chris** [comment]
That's interesting, could be specific to Chrome touchscreens. I definitely
can't check off Tasks in Habitica using touch on my Win10 machine, Chrome

**Comment by paglias** [comment]
Thanks for all the reports! Just wanted to let you know that we're aware of this bug and are investigating although so fare we've not been able to reproduce it on Chrome using touch input. 

I know it's difficult to get this info on a mobile device but it would  be very useful if someone manages to see if there's any error in the console after trying to complete a task

**Comment by b9chris** [comment]
Well again this occurs on any touchscreen with Chrome, and it's very easy to get a Dev Console on my Win10 laptop. I get no errors when attempting to complete a task with touch. It just behaves like mouseover. I highly suspect the existing click listener simply needs to also listen for the touch event.

The only stuff that does appear in Console is what I'm sure everyone sees on Chrome, the 2 warnings that appear upon loading Habitica:

moment.js:293 Deprecation warning: moment().zone is deprecated, use moment().utcOffset instead. http://momentjs.com/guides/#/warnings/zone/
Arguments: 
Error
    at g.zone (https://habitica.com/static/js/vendor.7d30b64844d224ff1067.js:6:44235)
    at Object.<anonymous> (https://habitica.com/static/js/app.574d28ed5c1add744845.js:1:291269)
    at n (https://habitica.com/static/js/manifest.1da315fd308b0a0541f8.js:1:101)
    at Object.<anonymous> (https://habitica.com/static/js/app.574d28ed5c1add744845.js:1:21394)
    at n (https://habitica.com/static/js/manifest.1da315fd308b0a0541f8.js:1:101)
    at Object.<anonymous> (https://habitica.com/static/js/app.574d28ed5c1add744845.js:1:406029)
    at n (https://habitica.com/static/js/manifest.1da315fd308b0a0541f8.js:1:101)
    at Object.<anonymous> (https://habitica.com/static/js/app.574d28ed5c1add744845.js:1:646710)
    at n (https://habitica.com/static/js/manifest.1da315fd308b0a0541f8.js:1:101)
    at Object.<anonymous> (https://habitica.com/static/js/app.574d28ed5c1add744845.js:1:398806)
x @ moment.js:293
(anonymous) @ moment.js:320
(anonymous) @ index.js:19
n @ bootstrap 368faddfa5b2d5da6b56:54
(anonymous) @ analytics.js:1
n @ bootstrap 368faddfa5b2d5da6b56:54
(anonymous) @ app.574d28ed5c1add744845.js:20701
n @ bootstrap 368faddfa5b2d5da6b56:54
(anonymous) @ menu.vue?8318:1
n @ bootstrap 368faddfa5b2d5da6b56:54
(anonymous) @ app.574d28ed5c1add744845.js:20005
n @ bootstrap 368faddfa5b2d5da6b56:54
(anonymous) @ app.vue?70c3:1
n @ bootstrap 368faddfa5b2d5da6b56:54
(anonymous) @ main.js:1
n @ bootstrap 368faddfa5b2d5da6b56:54
(anonymous) @ app.574d28ed5c1add744845.js:43839
n @ bootstrap 368faddfa5b2d5da6b56:54
window.webpackJsonp @ bootstrap 368faddfa5b2d5da6b56:25
(anonymous) @ app.574d28ed5c1add744845.js:1
4vuedraggable.umd.min.js:1 Options props is deprecated, add sortable options directly as vue.draggable item, or use v-bind. See https://github.com/SortableJS/Vue.Draggable/blob/master/documentation/migrate.md#options-props
created @ vuedraggable.umd.min.js:1
at @ vue.runtime.esm.js:1854
xe @ vue.runtime.esm.js:4213
t._init @ vue.runtime.esm.js:5002
a @ vue.runtime.esm.js:5148
Kt @ vue.runtime.esm.js:3283
init @ vue.runtime.esm.js:3114
l @ vue.runtime.esm.js:5972
u @ vue.runtime.esm.js:5919
h @ vue.runtime.esm.js:6047
u @ vue.runtime.esm.js:5948
h @ vue.runtime.esm.js:6047
u @ vue.runtime.esm.js:5948
(anonymous) @ vue.runtime.esm.js:6471
t._update @ vue.runtime.esm.js:3939
r @ vue.runtime.esm.js:4060
Ki.get @ vue.runtime.esm.js:4473
Ki @ vue.runtime.esm.js:4462
ge @ vue.runtime.esm.js:4067
He.$mount @ vue.runtime.esm.js:8409
init @ vue.runtime.esm.js:3118
l @ vue.runtime.esm.js:5972
u @ vue.runtime.esm.js:5919
h @ vue.runtime.esm.js:6047
u @ vue.runtime.esm.js:5948
h @ vue.runtime.esm.js:6047
u @ vue.runtime.esm.js:5948
h @ vue.runtime.esm.js:6047
u @ vue.runtime.esm.js:5948
(anonymous) @ vue.runtime.esm.js:6471
t._update @ vue.runtime.esm.js:3939
r @ vue.runtime.esm.js:4060
Ki.get @ vue.runtime.esm.js:4473
Ki @ vue.runtime.esm.js:4462
ge @ vue.runtime.esm.js:4067
He.$mount @ vue.runtime.esm.js:8409
init @ vue.runtime.esm.js:3118
l @ vue.runtime.esm.js:5972
u @ vue.runtime.esm.js:5919
h @ vue.runtime.esm.js:6047
u @ vue.runtime.esm.js:5948
h @ vue.runtime.esm.js:6047
u @ vue.runtime.esm.js:5948
x @ vue.runtime.esm.js:6210
O @ vue.runtime.esm.js:6313
x @ vue.runtime.esm.js:6191
O @ vue.runtime.esm.js:6313
(anonymous) @ vue.runtime.esm.js:6476
t._update @ vue.runtime.esm.js:3942
r @ vue.runtime.esm.js:4060
Ki.get @ vue.runtime.esm.js:4473
Ki.run @ vue.runtime.esm.js:4548
Oe @ vue.runtime.esm.js:4304
(anonymous) @ vue.runtime.esm.js:1980
lt @ vue.runtime.esm.js:1906
Promise.then (async)
xi @ vue.runtime.esm.js:1933
ct @ vue.runtime.esm.js:1990
Te @ vue.runtime.esm.js:4396
Ki.update @ vue.runtime.esm.js:4538
fi.notify @ vue.runtime.esm.js:730
set @ vue.runtime.esm.js:1055
(anonymous) @ generated:424
Promise.then (async)
created @ generated:423
at @ vue.runtime.esm.js:1854
xe @ vue.runtime.esm.js:4213
t._init @ vue.runtime.esm.js:5002
a @ vue.runtime.esm.js:5148
Kt @ vue.runtime.esm.js:3283
init @ vue.runtime.esm.js:3114
l @ vue.runtime.esm.js:5972
u @ vue.runtime.esm.js:5919
(anonymous) @ vue.runtime.esm.js:6510
t._update @ vue.runtime.esm.js:3939
r @ vue.runtime.esm.js:4060
Ki.get @ vue.runtime.esm.js:4473
Ki @ vue.runtime.esm.js:4462
ge @ vue.runtime.esm.js:4067
He.$mount @ vue.runtime.esm.js:8409
t._init @ vue.runtime.esm.js:5012
He @ vue.runtime.esm.js:5079
(anonymous) @ main.js:37
n @ bootstrap 368faddfa5b2d5da6b56:54
(anonymous) @ app.574d28ed5c1add744845.js:43839
n @ bootstrap 368faddfa5b2d5da6b56:54
window.webpackJsonp @ bootstrap 368faddfa5b2d5da6b56:25
(anonymous) @ app.574d28ed5c1add744845.js:1
Show 25 more frames

**Comment by SabreCat** [comment]
The linked tickets appear to all be variations of this. Will close the others and mark this to get help!

**Comment by kugoi** [comment]
Yeah, I still get mouse over behavior on task list. 

Sent from Yahoo Mail on Android 
 
  On Fri, Nov 1, 2019 at 1:33 PM, Sabe Jones<notifications@github.com> wrote:   
The linked tickets appear to all be variations of this. Will close the others and mark this to get help!

**Comment by b9chris** [comment]
Here's a video of me attempting to use a touchscreen to check off both To
Dos and Dailies in Chrome on Win10 (on a Microsoft Surface Book). No this
is not an ad for the salsa I accidentally included in the video ;)

https://youtu.be/yV_bkkb2nLU

**Comment by cvuorinen** [comment]
This has bugged me too for a long time. Finally got around to looking into it and found an easy solution. The SortableJS library that is used for the drag-n-drop sorting has options to add a delay on touch devices for the sort. Fixes both the scrolling issue and checking off tasks :tada:

**Comment by Alys** [comment]
@cvuorinen Awesome, thank you! I've marked this as in progress for you.
